### DISCUSSION TITLE
Duplicate a variant

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2305] [Ideas] [answered]
> Created by: Robin Wissink ([info@creativeorange.nl](info@creativeorange.nl)) on 2016/02/18 09:41:21 +0000
Votes at time of UserVoice import: 23

------
When managing a fashion store for example, you have to add a lot of variants for each product (colors, sizes, corresponding images). It would be great to be able to duplicate the first variant, and then for example only change the stock amount. This would greatly improve the experience of craft commerce for shop managers.

**Comment by lukeholder** [comment]
Related: https://github.com/craftcms/cms/issues/862

**Comment by lukeholder** [comment]
Related: #202

**Comment by jamiematrix** [comment]
Any updates/thoughts on this? We have a client who would like the same functionality - to be able to duplicate x copies of a variant within a product.

**Comment by aarmitage** [comment]
Just been with a client who have shown me a good use case for this. Duplicating a product obviously duplicates all variants, but then all stock fields, SKUs, possibly other attributes (colour, size etc) might also need to be amended. In short, copying a product only saves the creation of variants, but still amounts to the same amount of work to update all the existing fields. Fine if you're adding lots of T-Shirts all at once, not so helpful if your product list jumps around from t-shirt to pants to jackets and back to t-shirts.

Some fashion brands also use a common SKU with a suffix or just the last number that's different to represent a size or colour. From a workflow perspective, duplicating a single variant will copy over the correct SKU and any common attributes such as colour. It would then be quicker for managers to update a single digit on the SKU and not have to amend any common fields.

**Comment by wmdhosting** [comment]
+1 for this

**Comment by jelmer-scoremedia** [comment]
Would love to see all commerce entities, orders, order lines, discounts, variants etc.. to be implemented as elements for duplicating, relations.

**Comment by nfourtythree** [comment, marked as answer]
As of Commerce 5.1 it is now possible to duplicate variants.
